.github/workflows/e2e-detox-template.yml

```diff
@@ -107,13 +115,13 @@ jobs:
         with:
           parallelism: 10
           search_path: detox/e2e/test
-          device_name: ${{ steps.resolve-device.outputs.DEVICE_NAME }}
-          device_os_version: ${{ steps.resolve-device.outputs.DEVICE_OS_VERSION }}
+          device_name: ${{ env.DEVICE_NAME }}
+          device_os_version: ${{ env.DEVICE_OS_VERSION }}
 
   e2e-ios:
     name: ios-detox-e2e-${{ matrix.runId }}-${{ matrix.deviceName }}-${{ matrix.deviceOsVersion }}
     if: ${{ inputs.run-ios-tests }}
-    runs-on: macos-14
+    runs-on: macos-12
```

Comment:
Why this change ? macos-14 is faster and M1

Reply:
My Bad, I rebased and kept current changes